src/main/kotlin/racingcar/domain/Cars.kt

```diff
@@ -1,18 +1,12 @@
 package racingcar.domain
 
-import kotlin.streams.toList
-
-class Cars(var carList: List<Car> = emptyList()) {
-    fun moveCars() = carList.forEach() { it.move(Random().getRandomValue()) }
-
-    fun getWinner(): Car? = carList.maxByOrNull { it.position.position }
-
-    fun getWinnerList():
-        List<String> {
-            val winner = this.getWinner()
-            return carList.stream()
-                .filter { it.position.position == winner!!.position.position }
+class Cars(var cars: List<Car>) {
+    val winners: List<String>
+        get() {
+            return cars.filter { it -> it.position.position == cars.maxByOrNull { it.position.position }!!.position.position }
                 .map { it.name }
                 .toList()
         }
```

Comment:
```suggestion
        get() = cars.filter { it -> it.position.position == cars.maxByOrNull { it.position.position }!!.position.position }
                .map { it.name }
                .toList()
```

또는

```suggestion
        get() {
            val winningPosition = cars.maxByOrNull { it.position.position }!!.position.position

            return cars.filter { it -> it.position.position == winningPosition }
                .map { it.name }
                .toList()
        }
```

만약 함수를 전부 안으로 넣을 생각이면 expression 형식이 더 보기 좋을 것 같고, 만약 그렇지 않으면 따로 상수를 뽑는게 코드를 보기 더 편할 것 같아.

상수를 뽑는 것이 가지는 장점은 해당 반환값이 어떠한 역할을 하는지 좀 더 명확히 드러낼 수 있다거든.
그래서 단순히 코드 줄수를 줄이기 위해 코드 안에 다 넣기보단 이런 분리가 좀 더 코드가 잘 읽힐 수 있게 해주는 것 같아.

그리고 `!!` 연산자가 null이 반환될 경우 NPE를 던져준다고 알고 있는데 null이 반환되는 많은 로직들을 이런 방식으로 처리했더라고.

Kotlin이 Java에 비해 가지는 장점이 null 처리잖아. 그래서 null이 반환되는 로직에서 좀 더 명시적으로 처리를 해주면 좋을 것 같은데 이런 코드의 형식이면 Java 코드를 짜는 방식과 크게 다르지 않나?라는 생각이 들었어

이건 나도 확실한 부분이 아니라서 한번 생각해보는 것도 좋을 것 같아!
(근데 글을 쓰다보니 null이 반환될 수 있음을 알고, 그 상황에 대해서 개발자가 명시적으로 NPE를 던지는 거니까 괜찮은거 같기도?? ㅋㅋㅋ)

Reply:
(근데 글을 쓰다보니 null이 반환될 수 있음을 알고, 그 상황에 대해서 개발자가 명시적으로 NPE를 던지는 거니까 괜찮은거 같기도?? ㅋㅋㅋ) << - 이 부분에 대해서 어떠한 경우에도 null이 발생하지 않을걸 알기에 일종의 Assertion? 느낌으로 `!!` 를 사용하는거로 알고있었어 나는 ㅋㅋㅋ.. 이부분에 대한 정확한 공부가 필요할거같다 ㅋㅋ

코드 방식은 밑에가 더 잘 읽히는 코드인거같다 ㅎㅎ 변수로 분리하는게 아무래도 좋을 듯